confmap/provider/googlesecretsprovider/README.md

```diff
@@ -0,0 +1,59 @@
+# Google Secrets Provider
```

Comment:
Throughout this document, in place of `Otel` and `Otel Collector` let's just say `collector`. Additionally, the audience for this document is the user themselves, so we can use second party language like "you/your".